lua/roslyn/config.lua

```diff
@@ -142,9 +142,34 @@ function M.setup(user_config)
         table.insert(roslyn_config.args, "--stdio")
     end
 
+    -- filewatching: replace legacy boolean value (true -> auto; false -> force-nvim)
+    if type(roslyn_config.filewatching) == "boolean" then
+        if roslyn_config.filewatching then
+            roslyn_config.filewatching = "auto"
+        else
+            roslyn_config.filewatching = "force-nvim"
+        end
+        vim.notify(
+            "Value of the `filewatching` option should be 'auto' (default), 'force-nvim' or 'force-ls'.",
+            vim.log.levels.WARN,
+            { title = "roslyn.nvim" }
+        )
+    end
+
+    -- filewatching: disable client capability in 'force-ls' mode
+    if roslyn_config.filewatching == "force-ls" then
+        roslyn_config.config.capabilities = vim.tbl_deep_extend("force", roslyn_config.config.capabilities, {
+            workspace = {
+                didChangeWatchedFiles = {
+                    dynamicRegistration = false,
+                },
+            },
+        })
+    end
+
     -- HACK: Enable filewatching to later just not watch any files
     -- This is to not make the server watch files and make everything super slow in certain situations
-    if not roslyn_config.filewatching then
+    if roslyn_config.filewatching == "force-nvim" then
```

Comment:
```suggestion
    if roslyn_config.filewatching == "roslyn" or roslyn_config.filewatching == "off" then
```

The you can do `dynamicRegistration = roslyn_config.filewatching == "off"` on line 176

and then we don't need the block above